### ISSUE TITLE
multilib gcc fails to build

### ISSUE THREAD

**Issue by cpages** [NixOS/nixpkgs#8706]
There seems to be some multilib problem? Does anyone have a clue where to start looking / what might have changed to break this? I'm not really familiar with the fhsuserenv stuff.
```
building path(s) ‘/nix/store/r1ymf5ab44vyqknxcz97my8dmm48mq5p-gcc-4.9.3’, ‘/nix/store/ybjljnqfa54i3p6b500ggqbhilqyiisy-gcc-4.9.3-info’
$CPATH is `/nix/store/az2scrkb88l2q09xa0g6lpbv2mh8lxjl-zlib-1.2.8/include'
$LIBRARY_PATH is `/nix/store/az2scrkb88l2q09xa0g6lpbv2mh8lxjl-zlib-1.2.8/lib'
unpacking sources

...

checking for version 0.11 of ISL... yes
*** This configuration is not supported in the following subdirectories:
     target-libquadmath gnattools target-libada target-libgfortran target-libgo target-libffi target-libbacktrace target-libjava target-libobjc target-boehm-gc
    (Any other directories should still work fine.)
checking for default BUILD_CONFIG... bootstrap-debug
checking for --enable-vtable-verify... no
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: skipping incompatible /nix/store/8p0cgzn4azmn36fl29kkg9x17rmic4la-gcc-4.9.3/lib/gcc/x86_64-unknown-linux-gnu/4.9.3/libgcc.a when searching for -lgcc
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: cannot find -lgcc
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: skipping incompatible /nix/store/3vsv5di7216wp76limzkndynqrshfh86-glibc-multi-2.21/lib/libgcc_s.so when searching for -lgcc_s
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: skipping incompatible /nix/store/3vsv5di7216wp76limzkndynqrshfh86-glibc-multi-2.21/lib/libgcc_s.so when searching for -lgcc_s
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: skipping incompatible /nix/store/8p0cgzn4azmn36fl29kkg9x17rmic4la-gcc-4.9.3/lib/gcc/x86_64-unknown-linux-gnu/4.9.3/../../../../lib/libgcc_s.so when searching for -lgcc_s
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: skipping incompatible /nix/store/8p0cgzn4azmn36fl29kkg9x17rmic4la-gcc-4.9.3/lib/gcc/x86_64-unknown-linux-gnu/4.9.3/../../../libgcc_s.so when searching for -lgcc_s
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: cannot find -lgcc_s
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: skipping incompatible /nix/store/3vsv5di7216wp76limzkndynqrshfh86-glibc-multi-2.21/lib/libc.so when searching for -lc
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: skipping incompatible /nix/store/3vsv5di7216wp76limzkndynqrshfh86-glibc-multi-2.21/lib/libc.a when searching for -lc
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: skipping incompatible /nix/store/3vsv5di7216wp76limzkndynqrshfh86-glibc-multi-2.21/lib/libc.so when searching for -lc
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: skipping incompatible /nix/store/3vsv5di7216wp76limzkndynqrshfh86-glibc-multi-2.21/lib/libc.a when searching for -lc
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: cannot find -lc
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: skipping incompatible /nix/store/8p0cgzn4azmn36fl29kkg9x17rmic4la-gcc-4.9.3/lib/gcc/x86_64-unknown-linux-gnu/4.9.3/libgcc.a when searching for -lgcc
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: cannot find -lgcc
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: skipping incompatible /nix/store/3vsv5di7216wp76limzkndynqrshfh86-glibc-multi-2.21/lib/libgcc_s.so when searching for -lgcc_s
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: skipping incompatible /nix/store/3vsv5di7216wp76limzkndynqrshfh86-glibc-multi-2.21/lib/libgcc_s.so when searching for -lgcc_s
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: skipping incompatible /nix/store/8p0cgzn4azmn36fl29kkg9x17rmic4la-gcc-4.9.3/lib/gcc/x86_64-unknown-linux-gnu/4.9.3/../../../../lib/libgcc_s.so when searching for -lgcc_s
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: skipping incompatible /nix/store/8p0cgzn4azmn36fl29kkg9x17rmic4la-gcc-4.9.3/lib/gcc/x86_64-unknown-linux-gnu/4.9.3/../../../libgcc_s.so when searching for -lgcc_s
/nix/store/vgf2mygpwkc51cqbga6i60by3s5gb8dl-binutils-2.23.1/bin/ld: cannot find -lgcc_s
collect2: error: ld returned 1 exit status
configure: error: I suspect your system does not have 32-bit developement libraries (libc and headers). If you have them, rerun configure with --enable-multilib. If you do not have them, and want to build a 64-bit-only compiler, rerun configure with --disable-multilib.
note: keeping build directory ‘/tmp/nix-build-gcc-4.9.3.drv-0’
builder for ‘/nix/store/wvmq5m4bl0zjyqbqsrnxf1lhmqi64i0s-gcc-4.9.3.drv’ failed with exit code 1
cannot build derivation ‘/nix/store/z8c1g1yal8axig2zc2blzwd0i7p0ppld-steam-fhs.drv’: 1 dependencies couldn't be built
building path(s) ‘/nix/store/0bjzq08jsfbm123v0wyw6qg8m6gbpw10-unzip-6.0’
note: keeping build directory ‘/tmp/nix-build-unzip-6.0.drv-0’
cannot build derivation ‘/nix/store/bdx6ihwfg59naxxsnphw77zcqq5ji0lv-steam.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/bdx6ihwfg59naxxsnphw77zcqq5ji0lv-steam.drv’ failed
```

**Comment by cpages** [comment]
With this change to gcc, it builds and seems to create a proper env:
```
-    ${if enableMultilib then "--disable-libquadmath" else "--disable-multilib"}
+    ${if enableMultilib then "--enable-multilib --disable-libquadmath" else "--disable-multilib"}
```
However, it looks more a workaround than a proper fix, since --enable-multilib is the default, but the configure step seems to not find the proper headers/libs. @svanderburg any clue?

**Comment by vcunat** [comment]
Now this problem is in master, and we're about to release soon. There are only a few packages depending on it, but still.

Thanks a lot for this idea (I've spent quite some time on this before I found this report). I looked into configure.ac now, and this message seems triggered during some autodetection if multilib status isn't specified. I think yours is the proper solution. I'll do some more testing.
